### ISSUE TITLE
favicon should be in root path

### ISSUE THREAD

**Issue by sinkcup** [mkdocs/mkdocs#1962]
now mkdocs favicon is in `img/favicon.ico`, when accessing it in browser, there will be a 404 request of `/favicon.ico`.

I suggest mkdocs can support `/favicon.ico`.

demo: https://china-speed.org.cn/img/favicon.ico

![image](https://user-images.githubusercontent.com/4971414/72578712-d5a7e980-3911-11ea-93c1-52db6b1034fc.png)

**Comment by waylan** [comment]
Yes, browsers look in the root of a domain by default for a favicon. However, if you include a `link` in the `head` of your HTML, the browser will only check the path provided in that `link` instead. Like this:

```html
<link rel="shortcut icon" href="img/favicon.ico">
```

Of course, as all paths should be relative in MkDocs, the builtin themes do the following, which ensures the path is always relative to the current page after the template is rendered:

```django
<link rel="shortcut icon" href="{{ 'img/favicon.ico'|url }}">
```

That said, if you are building a custom theme, you can put the favicon anywhere you choose and provide a link in your templates to whatever path works for you. MkDocs assumes nothing about this. It is all controlled by the theme which is user customizable.

In other words:

1. The assertion that we must support favicons at the root is false, and...
2. we do support having the favicon elsewhere through theme customization.

**Comment by sinkcup** [comment]
@waylan Thanks, I have tried to change the favicon path when I am using themes, but there is a warning. it seems `site_favicon` was duplicated, so we can't change favicon path by config now?

```
site_favicon: "/favicon.ico"
```

> WARNING -  Config value: 'site_favicon'. Warning: Unrecognised configuration name: site_favicon

**Comment by waylan** [comment]
Correct. We deprecated that config setting a few versions ago as it was completely unnecessary. You can simply define the path within the template. No need to use a setting.

**Comment by sinkcup** [comment]
@waylan I am using a template not create a template, so I can't define it...

**Comment by waylan** [comment]
Then you can use [theme customization][1] and [template blocks][2] to override the part to the existing theme's template. Themes get to choose the default location of the favicon. Some third party themes use the root, others use their own custom location. Some might even provide there own custom theme specific config setting. If you don't like a given theme's choice, then don't use that theme or customize it.

As an aside, due to the bug reported in #1671, the current release doesn't allow you to override the theme file from the `docs_dir` even if you use the theme's chosen path. Until the next release provides a fix, the workaround is to use theme customization. 

[1]: https://www.mkdocs.org/user-guide/styling-your-docs/#using-the-theme-custom_dir
[2]: https://www.mkdocs.org/user-guide/styling-your-docs/#overriding-template-blocks
